lib/encoder.py

```diff
@@ -234,6 +248,10 @@ def __init__(self, option_set, string, formatter=OptionFormatter()):
       self._HandleFlag(match)
       unparsed = unparsed[match.end():]
       match = re.match(matcher, unparsed)
+    if option_set:
```

Comment:
When does an option set evaluate to false? It does seem to be mandatory from the constructor docstring, so I guess it can't be None. Does a false OptionSet mean it's empty? If so, won't also option_set.options simply be empty as well so the loop below simply gets zero iterations?